src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/DataSource.scala

```diff
@@ -60,6 +62,67 @@ class DataSource(graphFactory: OrientGraphFactory) extends LazyLogging {
 }
 
 class RawlsTransaction(graph: OrientGraph, dataSource: DataSource) {
+
+  var readLocks:mutable.Map[Element, Int] = mutable.Map[Element, Int]().withDefaultValue(0)
```

Comment:
Be a little careful with relying on the default implementations if you care about what's the underlying stuff as they can change, but I can only think of one important one over the last 3 releases of scala (and they changed back!)